internal/stores/contracts.go

```diff
@@ -424,12 +420,19 @@ func (s *SQLStore) Contracts() ([]rhpv2.Contract, error) {
 
 // RemoveContract implements the bus.ContractStore interface.
 func (s *SQLStore) RemoveContract(id types.FileContractID) error {
-	return s.db.Delete(&dbContractRHPv2{ID: id}).Error
+	return s.db.Transaction(func(tx *gorm.DB) error {
+		var contract dbContractRHPv2
+		if err := tx.Where(&dbContractRHPv2{FCID: id}).
+			Take(&contract).Error; err != nil {
+			return err
+		}
+		return tx.Where(&contract).Delete(&contract).Error
```

Comment:
I guess you have to find it before deleting it first here right? Just checking in the off-chance this was an oversight. I would've assumed you could do `Where(fcid = id).Delete(&contract{})`

Reply:
You can do that but then the `dbContract` won't be populated in `onBeforeDelete` hooks. Which is something my next PR is relying on atm.